starter-code/pizza.js

```diff
@@ -1,85 +1,200 @@
+	glutenFreeCrust: { name: "Gluten-free crust", price: 5 }
+};
 
 // Initial value of the state (the state values can change over time)
 var state = {
-  pepperonni: true,
-  mushrooms: true,
-  greenPeppers: true,
-  whiteSauce: false,
-  glutenFreeCrust: false
-}
+	pepperonni: true,
+	mushrooms: true,
+	greenPeppers: true,
+	whiteSauce: false,
+	glutenFreeCrust: true
+};
 
 // This function takes care of rendering the pizza based on the state
 // This function is triggered once at the begining and everytime the state is changed
 function renderEverything() {
-  renderPepperonni()
-  renderMushrooms()
-  renderGreenPeppers()
-  renderWhiteSauce()
-  renderGlutenFreeCrust()
-
-  renderButtons()
-  renderPrice()
+	renderPepperonni();
+	renderMushrooms();
+	renderGreenPeppers();
+	renderWhiteSauce();
+	renderGlutenFreeCrust();
+
+	renderButtons();
+	renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
-    if (state.pepperonni) {
-      $pep.style.visibility = "visible";
-    }
-    else {
-      $pep.style.visibility = "hidden";
-    }
-  })
+	document.querySelectorAll(".pep").forEach(function($pep) {
+		if (state.pepperonni) {
+			$pep.style.visibility = "visible";
+		} else {
+			$pep.style.visibility = "hidden";
+		}
+	});
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+	// Iteration 1: set the visibility of `<section class="mushroom">`
+	document.querySelectorAll(".mushroom").forEach(function($mush) {
+		if (state.mushrooms) {
+			$mush.style.visibility = "visible";
+		} else {
+			$mush.style.visibility = "hidden";
+		}
+	});
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+	// Iteration 1: set the visibility of `<section class="green-pepper">`
+	document.querySelectorAll(".green-pepper").forEach(function($green) {
+		if (state.greenPeppers) {
+			$green.style.visibility = "visible";
+		} else {
+			$green.style.visibility = "hidden";
+		}
+	});
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+	// Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+	document.querySelectorAll(".sauce").forEach(function($whiteSauce) {
+		if (state.whiteSauce) {
+			$whiteSauce.classList.add("sauce-white");
+		} else {
+			$whiteSauce.classList.remove("sauce-white");
+		}
+	});
 }
 
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+	// Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+	document.querySelectorAll(".crust").forEach(function($glutenFreeCrust) {
+		if (state.glutenFreeCrust) {
+			$glutenFreeCrust.classList.add("crust-gluten-free");
+		} else {
+			$glutenFreeCrust.classList.remove("crust-gluten-free");
+		}
+	});
 }
 
 function renderButtons() {
```

Comment:
So after some digging and consultation with the other TA's it was found that this can be done with a for loop:
Object.keys(state).forEach((item) => {
let btn = document.querySelector('.' + classMatchState[item]);
if (state[item]) {
btn.classList.add('active');
} else {
btn.classList.remove('active');
}
});

Kudos to Dalina I think.

However, you may have to create a separate if statement or something for green peppers, sauce and crust as the button class name is not equal to the state name.